### ISSUE TITLE
Jenkins jobs defined using generated DSL must specify all possible configuration for triggers, publishers, etc

### ISSUE THREAD

**Issue by g-dx** [jenkinsci/gitlab-plugin#613]
## Issue
Jenkins jobs defined using the `gitlab-plugin` generated DSL must specify all configuration. 

For example:

```
job('job-name') {

    scm {
        git('...some repo...')
    }

    triggers {
        gitlab {
            triggerOnPush(true)
            branchFilterType('NameBasedFilter')
            includeBranchesSpec 'master'
            triggerOnMergeRequest(false)
            triggerOnAcceptedMergeRequest(false)
            triggerOnClosedMergeRequest(false)
            triggerOpenMergeRequestOnPush('never')
            triggerOnNoteRequest(false)
            noteRegex('#rebuild')
            setBuildDescription(true)
            addNoteOnMergeRequest(false)
            skipWorkInProgressMergeRequest(true)
            ciSkip(true)
            addCiMessage(false)
            addVoteOnMergeRequest(false)
            acceptMergeRequestOnSuccess(false)
            excludeBranchesSpec('')
            targetBranchRegex('')
            mergeRequestLabelFilterConfig {
                include('')
                exclude('')
            }
            secretToken('')
        }
    }
    
    publishers {
        gitLabCommitStatusPublisher {
            name 'jenkins'
            markUnstableAsSuccess false
        }
        gitLabMessagePublisher {
            onlyForFailure true
            replaceSuccessNote false
            replaceFailureNote false
            replaceAbortNote false
            successNoteText ''
            failureNoteText ''
            abortNoteText ''
        }
    }

    steps {
        /* ... some steps ... */
    }
}

```
Otherwise executing the DSL script will fail.

For example the code above will fail when run against v1.4.7 of the `gitlab-plugin` with the following error:

```
Build step 'Invoke Gradle script' changed build result to SUCCESS
Processing DSL script job.groovy
ERROR: (job.groovy, line 8) the following options are required and must be specified: triggerOnAcceptedMergeRequest, triggerOnClosedMergeRequest
```
Due to the changes introduced in https://github.com/jenkinsci/gitlab-plugin/commit/a61a6c77be85b89a5f8b6d37cc382fb474ea5262.

Currently this means all new additions to the plugin DSL will cause _existing_ jobs to break when the `gitlab-plugin` is upgraded.

Is this a limitation of Jenkins when leveraging generated DSL functionality or can a feature request be considered to support using default values for configuration which is not specified?

### Context
- **Gitlab plugin version**: 1.4.5
- **Gitlab version**: 9.2.5
- **Jenkins version**: 2.46.3
- **Job type**: Freestyle

**Comment by g-dx** [comment]
After some research and consulting with other folks knowledgable in Jenkins I was pointed towards this discussion:

https://groups.google.com/forum/#!topic/job-dsl-plugin/JWO7zDuMYVA

I will open a pull request for this change.

**Comment by omehegan** [comment]
Thanks for any further info you can provide about this, we don't get too many issues related to Job DSL, I don't think it is as widely used anymore now that many folks are moving to Jenkins Pipeline. It does sound like a bad interaction between the two plugins.

**Comment by g-dx** [comment]
I've opened a pull request to make what I believe to be the necessary changes:

https://github.com/jenkinsci/gitlab-plugin/pull/616/

It would be great if you or someone could take a look and let me know if I'm on the right track, what I've missed, etc.

**Comment by omehegan** [comment]
This has just been released in plugin version 1.5.2. It will be in the update center within a few hours.
